packages/transformer/src/test/standalone/BranchProvenanceInitializer.test.ts

```diff
@@ -0,0 +1,287 @@
+    [[false, false, false], [2, 1]],
+    // "keep-reopened-db" is truthy but also equal to an optimized optional argument that we use
+    [[false, false, "keep-reopened-db"], [0, 0]],
+    [[false, true, false], [2, 0]],
+    [[false, true, "keep-reopened-db"], [0, 0]],
+    [[true, false, false], [1, 1]],
+    [[true, false, "keep-reopened-db"], [0, 0]],
+    [[true, true, false], [0, 0]],
+    [[true, true, "keep-reopened-db"], [0, 0]],
+  ]);
+
+  let index = 0;
+  // FIXME: don't use a separate iModel for each loop iteration, just add more element pairs
+  // to the one iModel. That will be much faster
+  for (const sourceHasFedguid of [true, false]) {
+    for (const targetHasFedguid of [true, false]) {
+      for (const createFedGuidsForMaster of ["keep-reopened-db", false] as const) {
+        it(`branch provenance init with ${[
+          sourceHasFedguid && "relSourceHasFedGuid",
+          targetHasFedguid && "relTargetHasFedGuid",
+          createFedGuidsForMaster && "createFedGuidsForMaster",
+        ].filter(Boolean)
+         .join(",")
+        }`, async () => {
+          let transformerMasterDb!: StandaloneDb;
+          let noTransformerMasterDb!: StandaloneDb;
+          let transformerForkDb!: StandaloneDb;
+          let noTransformerForkDb!: StandaloneDb;
+
+          try {
+            const suffixName = (s: string) => `${s}_${sourceHasFedguid ? "S" : "_"}${targetHasFedguid ? "T" : "_"}${createFedGuidsForMaster ? "C" : "_"}.bim`;
+            const sourceFileName = suffixName("ProvInitSource");
+            // const generatedIModel = generateIModel(sourceFileName, sourceHasFedguid, targetHasFedguid);
+            const sourcePath = IModelTransformerTestUtils.prepareOutputFile("IModelTransformer", sourceFileName);
+            if (fs.existsSync(sourcePath))
+              fs.unlinkSync(sourcePath);
+          
+            const generatedIModel = StandaloneDb.createEmpty(sourcePath, { rootSubject: { name: sourceFileName }});
+          
+            const physModelId = PhysicalModel.insert(generatedIModel, IModelDb.rootSubjectId, "physical model");
+            const categoryId = SpatialCategory.insert(generatedIModel, IModelDb.dictionaryId, "spatial category", {});
+          
+            const baseProps = {
+              classFullName: PhysicalObject.classFullName,
+              category: categoryId,
+              geom: IModelTransformerTestUtils.createBox(Point3d.create(1, 1, 1)),
+              placement: {
+                origin: Point3d.create(1, 1, 1),
+                angles: YawPitchRollAngles.createDegrees(1, 1, 1),
+              },
+              model: physModelId,
+            };
+            
+            const sourceFedGuid = sourceHasFedguid ? undefined : Guid.empty;
+            const sourceElem = new PhysicalObject({
+              ...baseProps,
+              code: Code.createEmpty(),
+              federationGuid: sourceFedGuid, // Guid.empty = 00000000-0000-0000-0000-000000000000
```

Comment:
I would remove the comment, sourceFedGuid is not always empty so this comment is invalid